internal/runbits/runtime/rationalize.go

```diff
@@ -0,0 +1,51 @@
+package runtime
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+func rationalizeError(auth *authentication.Auth, proj *project.Project, rerr *error) {
+	var errNoMatchingPlatform *model.ErrNoMatchingPlatform
+
+	isUpdateErr := errs.Matches(*rerr, &ErrUpdate{})
+	switch {
+	case rerr == nil:
+		return
+
+	case proj == nil:
+		multilog.Error("runtime:rationalizeError called with nil project, error: %s", errs.JoinMessage(*rerr))
+		*rerr = errs.Pack(*rerr, errs.New("project is nil"))
+
+	// Could not find a platform that matches on the given branch, so suggest alternate branches if ones exist
+	case isUpdateErr && errors.As(*rerr, &errNoMatchingPlatform):
+		branches, err := model.BranchNamesForProjectFiltered(proj.Owner(), proj.Name(), proj.BranchName())
+		if err == nil && len(branches) > 1 {
+			// Suggest alternate branches
+			*rerr = errs.NewUserFacingError(locale.Tr(
+				"err_alternate_branches",
+				errNoMatchingPlatform.HostPlatform, errNoMatchingPlatform.HostArch,
```

Comment:
nit: While I see the value in adding this suggestion it also adds complexity where we could just provide a more generic, but still helpful, message.

Reply:
This PR is just hoisting the existing error up to a user facing level. It's not seeking to make changes to the actual error.